virtual_rainforest/models/animals/model.py

```diff
@@ -0,0 +1,118 @@
+"""The :mod:`~virtual_rainforest.animal.model` module creates a
+:class:`~virtual_rainforest.animal.model.AnimalModel` class as a child of the
+:class:`~virtual_rainforest.core.model.BaseModel` class. At present a lot of the
+abstract methods of the parent class (e.g.
+:func:`~virtual_rainforest.core.model.BaseModel.setup` and
+:func:`~virtual_rainforest.core.model.BaseModel.spinup`) are overwritten using
+placeholder functions that don't do anything. This will change as the
+:mod:`virtual_rainforest` model develops. The factory method
+:func:`~virtual_rainforest.animals.model.AnimalModel.from_config` exists in a more
+complete state, and unpacks a small number of parameters from our currently pretty
+minimal configuration dictionary. These parameters are then used to generate a class
+instance. If errors crop here when converting the information from the config dictionary
+to the required types (e.g. :class:`~numpy.timedelta64`) they are caught and then
+logged, and at the end of the unpacking an error is thrown. This error should be
+caught and handled by downstream functions so that all model configuration failures
+can be reported as one.
+"""  # noqa: D205, D415
+
+
+from __future__ import annotations
+
+from typing import Any
+
+import pint
+from numpy import datetime64, timedelta64
+
+from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.core.model import BaseModel, InitialisationError
+
+
+class AnimalModel(BaseModel):
+    """A class describing the animal model.
+
+    Describes the specific functions and attributes that the animal module should
+    possess. Currently it is incomplete and mostly just a copy of the template set out
+    in AnimalModel.
+
+    Args:
+        update_interval: Time to wait between updates of the model state.
+
+
+    """
+
+    model_name = "animal"
+    """The model name for use in registering the model and logging."""
+
+    def __init__(
+        self,
+        update_interval: timedelta64,
+        start_time: datetime64,
+        **kwargs: Any,
+    ):
+
+        super().__init__(update_interval, start_time, **kwargs)
+
+    @classmethod
+    def from_config(cls, config: dict[str, Any]) -> AnimalModel:
+        """Factory function to initialise the animal model from configuration.
+
+        This function unpacks the relevant information from the configuration file, and
+        then uses it to initialise the model. If any information from the config is
+        invalid rather than returning an initialised model instance None is returned.
+
+        Args:
+            config: The complete (and validated) virtual rainforest configuration.
+
+        Raises:
+            InitialisationError: If configuration data can't be properly converted
+        """
+
+        # Assume input is valid until we learn otherwise
+        valid_input = True
+        try:
+            raw_interval = pint.Quantity(config["animal"]["model_time_step"]).to(
+                "minutes"
```

Comment:
I'm not sure about the choice of minutes as a unit here. Unless there's a good reason not to, I'd go with SI units everywhere to minimise confusion.

If you do want to go with a non-obvious unit, I'd incorporate this into the field name (i.e. `"model_time_step_minutes"`) to make it less ambiguous.

Reply:
That makes sense, if you're already using minutes elsewhere.

I'd open an issue for it and then fix it everywhere in one PR.

Reply:
More generally, the `int(round(raw_interval.magnitude))` patterns worries me: we shouldn't be silently changing settings. If someone had put in `1.5 minutes` then we shouldn't be chopping off those 30 seconds. If we don't want fractional time units, then we should block that in the config setup.

There is a problem here with using `pint` - it gives nice human readable units in the config, but we need to make sure that whatever comes in via `pint` gives an integer value on whatever our baseline temporal resolution is in `np.datetime64`. If we go for a fine enough baseline resolution then rounding is probably ok, but honestly I think we want to avoid rounding and it would be better to explicitly check that the value is a whole number of seconds.

Reply:
Agreed. You also don't need to cast to `int` if you're using `round` anyway.